python/lsst/cp/pipe/ptc.py

```diff
@@ -783,12 +928,12 @@ def calculateLinearityResidualAndLinearizers(self, exposureTimeVector, meanSigna
         dataset.polynomialLinearizerCoefficients : `list` of `float`
             Coefficients for LinearizePolynomial, where corrImage = uncorrImage + sum_i c_i uncorrImage^(2 +
             i).
-            c_(j-2) = -k_j/(k_1^j) with units (DN^(1-j)). The units of k_j are DN/t^j, and they are fit from
-            meanSignalVector = k0 + k1*exposureTimeVector + k2*exposureTimeVector^2 +...
-                             + kn*exposureTimeVector^n, with n = "polynomialFitDegreeNonLinearity".
-            k_0 and k_1 and degenerate with bias level and gain, and are not used by the non-linearity
-            correction. Therefore, j = 2...n in the above expression (see `LinearizePolynomial` class in
-            `linearize.py`.)
+            c_(j-2) = -k_j/(k_1^j) with units DN^(1-j) (c.f., Eq. 37 of 2003.05978). The units of k_j are
```

Comment:
It might be easier to put the full explanation in a docstring Notes, and keep the Returns list a bit simpler.
As mentioned above, putting everything necessary into the linearizer, and making the linearity type configurable will simplify this as well.